Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM golang:1.17.3
+
+ADD . cloud-api-adaptor
+RUN git clone -b CCv0-peerpod https://github.com/yoheiueda/kata-containers
+RUN apt-get update && apt-get install -y libvirt-dev libvirt0 genisoimage # other requirements?
+WORKDIR cloud-api-adaptor
+# generic build
+ENV CLOUD_PROVIDER=aws
+RUN make
+RUN install cloud-api-adaptor /usr/local/bin/cloud-api-adaptor
```

Comment:
nit: /usr/local/bin/cloud-api-adaptor -> /usr/local/bin/cloud-api-adaptor-aws

Reply:
It was like that because i implemented also the ibmcloud part, once other providers are supported they all can use this binary (except libvirt), i can change until then though

Reply:
cloud-api-adaptor binary is specific to the provider. So if we add additional provider binaries to the same image, then we need to change the binary name. If we create separate images for each provider then we can keep the binary name as-is.
I believe it's better to create single image per provider. 
What you can do is take the provider name as an argument 
Something like this

```
ARG CLOUD_PROVIDER
ENV CLOUD_PROVIDER=${CLOUD_PROVIDER}

RUN if [ "$CLOUD_PROVIDER" = "libvirt" ] ; then apt-get update -y && apt-get install -y libvirt-dev libvirt0 genisoimage; fi

...

Since the make is called with `CLOUD_PROVIDER=<value>` the same can be passed to run the docker build and you get the right image for the provider.

Reply:
Additionally the `CLOUD_PROVIDER` value can be suffixed to the image name. eg `quay.io/confidential-containers/cloud-api-adaptor-aws:<timestamp>`